src/components/file-system-explorer/file/index.tsx

```diff
@@ -35,12 +36,14 @@ const FileExplorer: FC<QuartzFileExplorerProps> = ({
   selected,
   isFileSelected,
   isValidExtension,
+  setIsEmptyFolder,
```

Comment:
this is a 3rd prop with a setter
have you considered Context here? We shouldn't be drilling props all the time imo

Reply:
Here I followed other props like "setActiveFile" and "setActiveFolder" that behaves the same way, I can look into how to optimise them together without breaking something later.